### PR TITLE
Feature/replicator db object

### PR DESCRIPTION
**Pull request by petevg**
This code is done-ish.

The tests don't consistently run, when run as a suite, possibly due to https://cloudant.fogbugz.com/f/cases/29000/Replication-doesn-t-trigger-after-creating-replication-job-in-Dashboard

Pushing so that I can reference the code.

@evansde77 @ksnavely @jcounts @dmannarino

### CODE REVIEW

File: src/cloudant/database.py [hunk 1]
```diff
@@ -24,23 +24,35 @@ def __init__(self, cloudant, database_name, fetch_limit=100):
...
-            )
+    @property
+    def database_url(self):
+        return posixpath.join(
+            self._database_host,
+            urllib.quote_plus(self._database_name)
         )
 
+    @property
+    def creds(self):
```
Comment (evansde77) [inline src/cloudant/database.py]: I think I would go for a more descriptive name here, like build_creds or something like that, but NBD

File: src/cloudant/replicator.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        @param str replication_id: id of the replication to inspect.
+
+        """
+        try:
+            repl_doc = self[repl_id]
+        except KeyError:
+            raise CloudantException(
+                "Replication {} not found".format(replication_id))
+        repl_doc.fetch()
+        print "repl_doc: {}".format(repl_doc)
```
Comment (evansde77) [inline src/cloudant/replicator.py]: logger?

**Comment by evansde77** [comment]
+1

File: src/cloudant/replicator.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        """
+        _list_replications_
+
+        Returns a list of all replications.
+
+        """
+        docs = self.all_docs(include_docs="true")['rows']
+        return [doc['doc'] for doc in docs]
+
+    def replication_state(self, repl_id):
```
Comment (jcounts) [inline src/cloudant/replicator.py]: Do we want to do anything in the case where a replication document exists (has been posted), but has yet to trigger? In this case _replication_state would not exist.

I may just be overthinking things...

File: src/cloudant/replicator.py (hunk 2 above)
Comment (petevg) [inline src/cloudant/replicator.py]: Whoops. Meant to remove that (artifact of being very puzzled as to why stuff was failing.)

File: src/cloudant/replicator.py (hunk 3 above)
Comment (petevg) [inline src/cloudant/replicator.py]: jcounts: in that case, we return None (and the test verifies that we do ... or, at least, considers that a valid return). When I think about couch docs, I generally think in JavaScript, so I think of a nonexistent field as simply having a null value, and "None" reflects that in Python.

We could raise an Exception, but that feels less correct to me ...

**Pull request merged**